### DISCUSSION TITLE
ColorPicker

### DISCUSSION THREAD

**Discussion by jocelynnatali** [tui-cs/Terminal.Gui#1663] [Ideas]
I would like develop a ColorPicker control. It could be embedded or not in a dialogue.
At first, it could only manager 2 or 16 colors selection and maybe later true colors.
What do you think about that ?

![ColorPicker](https://user-images.githubusercontent.com/2706720/163322857-8d48d818-292c-42e1-b071-dbd3357c0f5b.png)

**Comment by tznind** [comment]
Thanks for getting involved.  That sounds like a cool control to have.  The normal workflow for contributing is:

- Create new control
- Create a Scenario in UICatalog that demonstrates how to use the control
- Add unit tests for the control
- Open a Pull Request

Controls should work with keyboard only and then also mouse (if possible).

How do you intend to show the current selection? Just rely on the console cursor or add a rune (e.g. bullet) for the selected box?

This is the 'color picker' I created for [TerminalGuiDesigner](https://github.com/tznind/TerminalGuiDesigner/blob/main/src/UI/Windows/ColorPicker.cs).  I just used radio buttons since I only needed the user to pick colors from the gui.cs `Color` Enum.  I'm not sure if you were talking about that or trying to implement something that picks Hex colours too (e.g. #FF00FF).  Just showing mine as a reference, I think your dialog looks cleaner and nicer if it can be made to work :) 

![image](https://user-images.githubusercontent.com/31306100/163342707-6e6bc55e-37a8-4638-b5cf-0ff6b3ead750.png)

**Reply by jocelynnatali** [reply]
Thanks for the contributing workflow explanations. Your projet is really cool.

For the seleted color, we can display this character ░ (2591) and white or black for the foreground and the enum color on the background.
We could use the keyboard arrows to select the right color or use the mouse.

![image](https://user-images.githubusercontent.com/2706720/163443758-9372b73f-e926-4359-bc14-9b456d33ce58.png)

For the monochrome display, it can just display the 2 colors in the grid.
![image](https://user-images.githubusercontent.com/2706720/163448443-0e6f1d52-2c35-4d41-aef5-1e1afff772bf.png)

For true color, I don't know yet. Maybe 256 is more possible with 3 * 8 values.

**Comment by BDisp** [comment]
Thanks for your effort @jocelynnatali, i think would be great. For a true color you can also see this PR #1628 that still in progress.

**Reply by jocelynnatali** [reply]
Thank you for the link :-).

**Reply by BDisp** [reply]
I think still missing the true color solution for linux.

**Comment by jocelynnatali** [comment]
Ok. So, I think I fucus on 2 and 16 colors first and maybe 256 later.


Télécharger Outlook pour Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by jocelynnatali** [comment]
The PR is here : https://github.com/migueldeicaza/gui.cs/pull/1668 :-)
